doop-central/ui.go

```diff
@@ -59,45 +59,12 @@ func (ui UI) RenderMainPage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	reports, err := ui.d.GetReports()
+	data, err := retrieveData(ui.downloader, ui.docstrings, (r.URL.Path == "/all"))
```

Comment:
The `retrieveData` and `showAll` arguments do not change the behavior of the helper function, they are just entered into the `data` struct. So this concern can be removed from the helper function entirely. Then, with `ui.downloader` being the last remaining argument, it would make semantic sense to make `retrieveData` a member function of `*Downloader`.
```suggestion
	data, err := ui.downloader.retrieveData()
	data.Docstrings = ui.docstrings
	data.ShowAll = r.URL.Path == "/all"
```